### PR TITLE
chore: Makefile full schema versioning

### PR DESCRIPTION
**Pull request by toddbaert**
Adds make target that pulls the specified version of the flagd-json-schema and serves it at the appropriate location

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -125,11 +126,12 @@ generate-proto-docs: pull-schemas-submodule
 	&& sed '/^## Table of Contents/,/#top/d' ${PWD}/$(DOCS_DIR)/reference/specifications/protos-with-toc.md >> ${PWD}/$(DOCS_DIR)/reference/specifications/protos.md \
 	&& rm -f ${PWD}/$(DOCS_DIR)/reference/specifications/protos-with-toc.md
 
-# Update the schema at flagd.dev
-# PUBLIC_JSON_SCHEMA_DIR above controls the dir (and therefore major version)
+# update or add the publicly served schema at https://flagd.dev/schema/${JSON_SCHEMA_VERSION}/flags.json
+# update JSON_SCHEMA_VERSION var at the top of this file with a the version to be updated or addeds
```
Comment (toddbaert) [inline Makefile:130]:
```suggestion
# update JSON_SCHEMA_VERSION var at the top of this file with the version to be updated or added and served publicly
```

File: Makefile [hunk 2]
```diff
@@ -1,6 +1,7 @@
 PHONY: .docker-build .build .run .mockgen
 PREFIX=/usr/local
-PUBLIC_JSON_SCHEMA_DIR=docs/schema/v0/
+JSON_SCHEMA_VERSION=v0.2.3
```
Comment (Kavindu-Dodan) [inline Makefile:3]: can this be something for renovate based auto updates? [1] 🤔 

[1] - https://github.com/open-feature/flagd/blob/main/renovate.json

Comment (toddbaert) [reply Makefile:3]: I was thinking about that but I decided not to. My reasoning is that in many cases we will only want to serve a new schema with some thought and consideration, and probably at least an implementation in flagd (like for example @bacherfl change for weighing)

Comment (toddbaert) [reply Makefile:3]: Yes. I could also take it from an env var and just default it here. You only need to run it once and it creates the dirs (which correspond to the served URLs)

Comment (Kavindu-Dodan) [reply Makefile:3]: This is fine, we just need to remember the change :)

**Pull request closed without merging**